pkg/agentconfig/container.go

```diff
@@ -26,15 +26,35 @@ func (a *ContainerBuilder) AgentContainer(ctx context.Context) (*core.Container,
 	ports := make([]core.ContainerPort, 0, 5)
 	confCns := a.configuredContainers(ctx)
 
+	names := make(map[string]int)
 	a.eachConfiguredContainer(confCns, func(app *core.Container, cc *Container) {
 		switch cc.Replace {
 		case ReplacePolicyContainer:
 			// Simply inherit the ports of the replaced container
 			ports = append(ports, app.Ports...)
 		case ReplacePolicyIntercept:
 			for _, ic := range PortUniqueIntercepts(cc) {
+				name := ic.ContainerPortName
+
+				if _, ok := names[name]; ok {
```

Comment:
Integration tests revealed a flaw. Unnamed ports must not be subjected to the duplication avoidance logic. As things stands now, the first unnamed port is OK, but the second is named "-2", and that's not a valid port name.

Reply:
I believe this is fixed.